team-manager/src/main/java/ua/knu/knudev/teammanager/domain/Department.java

```diff
@@ -0,0 +1,36 @@
+package ua.knu.knudev.teammanager.domain;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.UuidGenerator;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(schema = "team_management", name = "department")
+public class Department {
+
+    @Id
+    @UuidGenerator
+    @Column(updatable = false, nullable = false)
+    private UUID id;
+
+    @Column(nullable = false)
+    private String name;
+
+    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.PERSIST)
+    @JoinTable(name = "departments_specialties",
+            joinColumns = @JoinColumn(name = "departments"),
+            inverseJoinColumns = @JoinColumn(name = "specialties"))
+    private Set<Specialty> specialties = new HashSet<>();
+
+}
```

Comment:
Add here helper method
` public void addSpecialty(Specialty specialty) {
        this.specialties.add(specialty);
        specialty.getDepartments().add(this);
    }`

by this method you will have bidirectional mapping